# We use a 5x5 adjacent matrix to represent the communication graph between robots.
# This is a 1x5 vector represents the communication status of one robot.
bool[5] state